RevenueCat/Scripts/Purchases.cs

```diff
@@ -1271,6 +1271,72 @@ public void RedeemWebPurchase(WebPurchaseRedemption webPurchaseRedemption, Redee
         _wrapper.RedeemWebPurchase(webPurchaseRedemption);
     }
 
+    public delegate void GetEligibleWinBackOffersForProductFunc(WinBackOffer[] winBackOffers, Error error);
+
+    private GetEligibleWinBackOffersForProductFunc GetEligibleWinBackOffersForProductCallback { get; set; }
+
+    /// <summary>
+    /// Gets eligible win-back offers for a given store product. Only available on iOS 18.0+ with StoreKit 2.
+    /// Returns an error if the platform is not iOS 18.0+ or if StoreKit 2 is not used.
+    /// </summary>
+    /// <param name="storeProduct">The Purchases.StoreProduct to get win-back offers for</param>
+    /// <param name="callback">A callback receiving an array of Purchases.WinBackOffer objects or an error if unsuccessful</param>
+    public void GetEligibleWinBackOffersForProduct(StoreProduct storeProduct, GetEligibleWinBackOffersForProductFunc callback)
+    {
+        GetEligibleWinBackOffersForProductCallback = callback;
+        _wrapper.GetEligibleWinBackOffersForProduct(storeProduct);
+    }
+
+    public delegate void GetEligibleWinBackOffersForPackageFunc(WinBackOffer[] winBackOffers, Error error);
+
+    private GetEligibleWinBackOffersForPackageFunc GetEligibleWinBackOffersForPackageCallback { get; set; }
+
+    /// <summary>
+    /// Gets eligible win-back offers for a given package. Only available on iOS 18.0+ with StoreKit 2. 
+    /// Returns an error if the platform is not iOS 18.0+ or if StoreKit 2 is not used.
+    /// </summary>
+    /// <param name="package">The Purchases.Package to get win-back offers for</param>
+    /// <param name="callback">A callback receiving an array of Purchases.WinBackOffer objects or an error if unsuccessful</param>
+    public void GetEligibleWinBackOffersForPackage(Package package, GetEligibleWinBackOffersForPackageFunc callback)
+    {
+        GetEligibleWinBackOffersForPackageCallback = callback;
+        _wrapper.GetEligibleWinBackOffersForPackage(package);
+    }
+
+    public delegate void PurchaseProductWithWinBackOfferFunc(string productIdentifier, CustomerInfo customerInfo, bool userCancelled, Error error);
+
+    private PurchaseProductWithWinBackOfferFunc PurchaseProductWithWinBackOfferCallback { get; set; }
+
+    /// <summary>
+    /// Purchase a product with a win-back offer. Only available on iOS 18.0+ with StoreKit 2.
+    /// Returns an error if the platform is not iOS 18.0+ or if StoreKit 2 is not used.
+    /// </summary>
+    /// <param name="storeProduct">The Purchases.StoreProduct to purchase</param>
+    /// <param name="winBackOffer">The Purchases.WinBackOffer to use</param>
+    /// <param name="callback">A callback receiving the product identifier, customer info, user cancellation, and an error if the purchase fails</param>
+    public void PurchaseProductWithWinBackOffer(StoreProduct storeProduct, WinBackOffer winBackOffer, PurchaseProductWithWinBackOfferFunc callback)
```

Comment:
The other PurchaseProduct function accepts the `productIdentifier` as a string, but here I chose to accept the `StoreProduct` since that's more consistent with what we do in other places. 

Can y'all think of any reason that this might should accept the productIdentifier instead?